clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -2252,6 +2252,30 @@ static mlir::Value emitNeonRShiftImm(CIRGenFunction &cgf, mlir::Value shiftVec,
                              false /* right shift */);
 }
 
+/// Vectorize value, usually for argument of a neon SISD intrinsic call.
+static void vecExtendIntValue(CIRGenFunction &cgf, cir::VectorType argVTy,
+                              mlir::Value &arg, mlir::Location loc) {
+  CIRGenBuilderTy &builder = cgf.getBuilder();
+  cir::IntType eltTy = mlir::dyn_cast<cir::IntType>(argVTy.getEltType());
+  assert(mlir::isa<cir::IntType>(arg.getType()) && eltTy);
+  // The constant argument to an _n_ intrinsic always has Int32Ty, so truncate
+  // it before inserting.
+  arg = builder.createIntCast(arg, eltTy);
+  mlir::Value zero = builder.getConstInt(loc, cgf.SizeTy, 0);
+  arg = builder.create<cir::VecInsertOp>(
+      loc, builder.create<cir::PoisonOp>(loc, argVTy), arg, zero);
```

Comment:
Can't we just assume poison during LLVM lowering and avoid introducing a new op? See https://discourse.llvm.org/t/please-dont-use-undef-in-tests-part-2/83388/11

Reply:
The difficulty is that we'd have to introduce an attribute to cir::VecInsertOp to say the vector argument is poison value. I feel it'd be an overkill given most of use cases of cir::VecInsertOp wouldn't have Poison value as argument. 
This case is kinda really unique.